scripts/check-type-dependencies.js

```diff
@@ -86,11 +86,8 @@ function checkTypes(pkg) {
     'utf8',
   );
   const deps = (typeDecl.match(/from '.*'/g) || [])
-    .map(match => match.replace(/^.*from '(.*)'.*$/, '$1'))
-    .filter(
-      n =>
-        !(n.startsWith('.') || n === pkg.name || n.startsWith(`${pkg.name}/`)),
-    );
+    .map(match => match.replace(/from '(.*)'/, '$1'))
+    .filter(n => !n.startsWith('.'));
```

Comment:
do you think it was a bug to even have `n === pkg.name` in there? i can understand the last one, but that would still be ok right, if perhaps a bit redundant?

Reply:
It should never be importing itself since it's all rolled up into one file